V3.0/UFTP-V3.0-metering.xsd

```diff
@@ -0,0 +1,157 @@
+                </xs:annotation>
+            </xs:enumeration>
+        </xs:restriction>
+    </xs:simpleType>
+    <!-- -->
+    <!--  Metering Message Type -->
+    <!-- -->
+    <xs:complexType name="MeteringMessageType">
+        <xs:complexContent>
+            <xs:extension base="PayloadMessageType">
+                <xs:sequence>
+                    <xs:annotation>
+                        <xs:documentation>The message must contain one or more profiles with metering, energy or price values.</xs:documentation>
+                    </xs:annotation>
+                    <xs:element name="Profile" type="MeteringProfileType" maxOccurs="unbounded"/>
+                </xs:sequence>
+                <xs:attribute name="Revision" type="xs:long" use="required">
+                    <xs:annotation>
+                        <xs:documentation>Revision of this message. A sequence number that must be incremented each time a new revision of a metering message is sent.</xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="ISP-Duration" type="xs:duration" use="required">
+                    <xs:annotation>
+                        <xs:documentation>ISO 8601 time interval (minutes only, for example PT15M) indicating the duration of the ISPs referenced in this message. Although the ISP length is a market-wide fixed value, making this assumption explicit in each message is important for validation purposes, allowing implementations to reject messages with an errant ISP duration.</xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="TimeZone" type="TimeZoneNameType" use="required">
+                    <xs:annotation>
+                        <xs:documentation>Time zone ID (as per the IANA time zone database, http://www.iana.org/time-zones, for example: Europe/Amsterdam) indicating the UTC offset that applies to the Period referenced in this message. Although the time zone is a market-wide fixed value, making this assumption explicit in each message is important for validation purposes, allowing implementations to reject messages with an errant UTC offset.</xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="Currency" type="ISO4217CurrencyType" use="optional">
+                    <xs:annotation>
+                        <xs:documentation>ISO 4217 code indicating the currency that applies to the price of the Tariff Rates. Only required if ImportTariff or ExportTariff profiles are included.</xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="Period" type="PeriodType" use="required">
+                    <xs:annotation>
+                        <xs:documentation>Day (in yyyy-mm-dd format) the ISPs referenced in this Metering message belong to.</xs:documentation>
```

Comment:
can you make yyyy-mm-dd all-caps: YYYY-MM-DD ? Prevents any confusion with m as minutes in the ISO standard.

Reply:
done